server/datastore/mysql/scripts.go

```diff
@@ -24,20 +24,20 @@ func (ds *Datastore) NewHostScriptExecutionRequest(ctx context.Context, request
 		var err error
 		if request.ScriptContentID == 0 {
 			// then we are doing a sync execution, so create the contents first
-			scRes, err := insertScriptContents(ctx, request.ScriptContents, tx)
+			scRes, err := insertScriptContents(ctx, tx, request.ScriptContents)
 			if err != nil {
 				return err
 			}
 
 			id, _ := scRes.LastInsertId()
 			request.ScriptContentID = uint(id)
 		}
-		res, err = newHostScriptExecutionRequest(ctx, request, tx)
+		res, err = newHostScriptExecutionRequest(ctx, tx, request)
 		return err
 	})
 }
 
-func newHostScriptExecutionRequest(ctx context.Context, request *fleet.HostScriptRequestPayload, tx sqlx.ExtContext) (*fleet.HostScriptResult, error) {
+func newHostScriptExecutionRequest(ctx context.Context, tx sqlx.ExtContext, request *fleet.HostScriptRequestPayload) (*fleet.HostScriptResult, error) {
```

Comment:
Moved the `tx` to second arg after `ctx` as we have static linting rules that check for those conventions and make sure that `ds.reader` or `ds.writer` are not used inside those functions.